tasks/DroneFlightControlTask.cpp

```diff
@@ -0,0 +1,1456 @@
+   * 7 - RTK Detailed Information
+   * 8 - Magnetometer
+   * 9 - RC Channels Data
+   * 10 - Gimbal Data
+   * 11 - Flight Status
+   * 12 - Battery Level
+   * 13 - Control Information
+   */
+    freq[0] = DataBroadcast::FREQ_HOLD;
+    freq[1] = DataBroadcast::FREQ_HOLD;
+    freq[2] = DataBroadcast::FREQ_HOLD;
+    freq[3] = DataBroadcast::FREQ_HOLD;
+    freq[4] = DataBroadcast::FREQ_HOLD;
+    freq[5] = DataBroadcast::FREQ_50HZ; // This is the only one we want to change
+    freq[6] = DataBroadcast::FREQ_HOLD;
+    freq[7] = DataBroadcast::FREQ_HOLD;
+    freq[8] = DataBroadcast::FREQ_HOLD;
+    freq[9] = DataBroadcast::FREQ_HOLD;
+    freq[10] = DataBroadcast::FREQ_HOLD;
+    freq[11] = DataBroadcast::FREQ_HOLD;
+    freq[12] = DataBroadcast::FREQ_HOLD;
+    freq[13] = DataBroadcast::FREQ_HOLD;
+
+    ACK::ErrorCode ack = mSetup.vehicle->broadcast->setBroadcastFreq(freq, 1);
+    if (ACK::getError(ack))
+    {
+        ACK::getErrorCodeMessage(ack, __func__);
+        return false;
+    }
+    else
+    {
+        return true;
+    }
+}
+
+ErrorCode::ErrorCodeType DroneFlightControlTask::runWaypointV2Mission()
+{
+    if (!mSetup.vehicle->isM300())
+    {
+        DSTATUS("This sample only supports M300!");
+        return false;
+    }
+
+    GetRemainRamAck actionMemory = {0};
+    ErrorCode::ErrorCodeType ret;
+
+    /*! upload mission */
+    /*! upload mission's timeout need to be longer than 2s*/
+    int uploadMissionTimeOut = 3;
+    ret = uploadWaypointMission(uploadMissionTimeOut);
+    if (ret != ErrorCode::SysCommonErr::Success)
+        return ret;
+    sleep(mFunctionTimeout);
+
+    /*! download mission */
+    std::vector<WaypointV2> mission;
+    ret = downloadWaypointMission(mission);
+    if (ret != ErrorCode::SysCommonErr::Success)
+        return ret;
+    sleep(mFunctionTimeout);
+
+    /*! upload  actions */
+    /*! check action memory */
+    ret = getActionRemainMemory(actionMemory);
+    if (actionMemory.remainMemory <= 0)
+    {
+        DSTATUS("action memory is not enough.Can not upload more action!");
+        return ErrorCode::SysCommonErr::UndefinedError;
+    }
+
+    ret = uploadWapointActions();
+    if (ret != ErrorCode::SysCommonErr::Success)
+        return ret;
+
+    ret = getActionRemainMemory(actionMemory);
+    sleep(mFunctionTimeout);
+
+    /*! start mission */
+    ret = startWaypointMission();
+    if (ret != ErrorCode::SysCommonErr::Success)
+        return ret;
+    sleep(20);
+
+    /*! set global cruise speed */
+    setGlobalCruiseSpeed(1.5);
+    sleep(mFunctionTimeout);
+
+    /*! get global cruise speed */
+    getGlobalCruiseSpeed();
+    sleep(mFunctionTimeout);
+
+    /*! pause the mission*/
+    ret = pauseWaypointMission();
+    if (ret != ErrorCode::SysCommonErr::Success)
+        return ret;
+    sleep(5);
+
+    /*! resume the mission*/
+    ret = resumeWaypointMission();
+    if (ret != ErrorCode::SysCommonErr::Success)
+        return ret;
+    sleep(50);
```

Comment:
if I'm understanding this correctly, would every waypoint mission take 20sec to start, then a couple of seconds to set and print the global cruise speed, 5 more seconds paused, then 5 more seconds after the pause, and finally, 50 sec to complete the mission before returning? Isnt it better to calculate the expected time to complete the mission? I dont see it always taking 50 seconds no matter the mission.